x-pack/plugins/security_solution/public/management/common/polling.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const POLL_INTERVAL = 10000;
+
+export const startPoll = ({
+  pollAction,
+  interval,
+  shouldStop,
+  stopAction,
+}: {
+  pollAction: () => void;
+  interval: number;
+  shouldStop: () => boolean;
+  stopAction: () => void;
+}) => {
+  function sleep() {
+    return new Promise((resolve) => setTimeout(resolve, interval));
```

Comment:
we'll still need to read in conditions for `shouldStop()` so that we can stop the `setInterval()` from restarting.  I think it will be important to control when this poll should stop in an explicit way.  Same with `stopAction()`, I'd like to have an ability to perform some action after stopping a poll.

I think you're suggesting using `setInterval()` directly as the polling mechanism and then making decisions to start/stop  directly in the middleware?

Reply:
The way I understand it because of using setTimeout you don't really have explicit handle to the polling process (because setTimeout will be invoked multiple time producing different handles), because of that you have to pass those two callbacks (shouldStop and stopAction) which makes the code flow quite indirect to be honest (it took me a while to follow it and I still don't have a complete picture of the flow). By using setInterval you get a single handle to the interval that could be returned and stored in the redux and used to stop the polling directly from redux when setting flag to false. I would think it's also a little bit less resource intensive as you don't create multiple promises and multiple callbacks.

Another suggestion could be not to stop the interval itself at all. Start it once during the redux creation and just skip refresh based on the flag. I don't think we are talking about very resource intensive thing - checking a condition once every 10 seconds should not be performance concern to my opinion.

One thing I'm a bit surprised though that such a common Kibana functionality doesn't have a specific API/solution/pattern in Kibana architecture.